### DISCUSSION TITLE
Is there a way to add a list of emails into the segment?

### DISCUSSION THREAD

**Discussion by jibin2706** [Flagsmith/flagsmith#840] [Q&A] [answered]
I can have a bunch of email addresses, that have to be added to a segment. Is there a better way to do this?
<img width="767" alt="image" src="https://user-images.githubusercontent.com/28289448/156893180-c05eca51-a8a8-457b-aadf-b3ccc49590a5.png">

**Comment by dabeeeenster** [comment, marked as answer]
You can do `contains` and then separate the email addresses with a character thats not a legal email address character like a comma.

**Reply by jibin2706** [reply]
Is there a better way, the list could have hundreds of emails. Managing it would be difficult.

**Reply by dabeeeenster** [reply]
Is there a canonical list of these email addresses somewhere? If it was a larger number we would recommend using a common trait and setting it in the Flagsmith SDK for all those emails.

**Reply by jibin2706** [reply]
The email list can keep on changing between features. (Beta users per feature)

**Reply by dabeeeenster** [reply]
How do they opt in to the beta test? Is this data stored anywhere else?

**Reply by jibin2706** [reply]
Accepted on a form. And can be added later by us.

**Reply by dabeeeenster** [reply]
You could use the Flagsmith API, setting a trait for users that register via the form?

**Reply by jibin2706** [reply]
We manually add users in batches. I think using the `contains` on the email list is the solution for us.

**Reply by dabeeeenster** [reply]
The other option if you have the list available within the context of your app is set a `Trait` to define that `Identity` as part of the list and then just make a simple rule on that `Trait` value
